### ISSUE TITLE
String criteria unnecessarily being converted to RegExp

### ISSUE THREAD

**Issue by bradshja** [balderdashy/sails-mongo#302]
It seems like the where parser is being a little too aggressive with converting string criteria into regular expressions.  This is normally not a problem, but we have a database with over 1 million devices in it, and queries against that table using the MAC address are very slow.  Passing the MAC address as an array makes it perform over 50 times faster.

The reason for this appears to be Query.parseValue converting strings to regular expressions even when there is no reason to, which it doesn't appear to do for arrays of strings.  Shouldn't this only be necessary if there are wildcards that need to be searched against?  If I add a check for that before converting to a regular expression, performance is much faster.

I'm submitting a PR with a possible fix momentarily.

**Comment by bradshja** [comment]
I submitted PR #303 to fix this, but regrettably it fails the automated tests for reasons that seem unrelated to the code changes.  I don't know enough about how the CI tests are run to fix it right now, but if I find time I'll dig into it.

**Comment by mofux** [comment]
+1 If there wasn't this RegExp conversion it would also allow us to pass a mongo $where query like

```javascript
{
  '$where': 'this.price > this.retailPrice + 100'
}
```

... although it may be cleaner to solve this particular query type somewhere else in the code...

**Comment by sailsbot** [comment]
Thanks for posting, @bradshja.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
